examples/log-ingestion-k8s/README.md

```diff
@@ -0,0 +1,102 @@
+   $ kubectl create -f my-app-03-deployment.yaml
+   ```
+   2. Deploy FluentBit as DaemonSet in `logging` namespace:
+   ```
+   $ kubectl apply -f fluent-bit-01-ns.yaml
+   $ kubectl create -f fluent-bit-02-service-account.yaml
+   $ kubectl create -f fluent-bit-03-role-1.22.yaml
+   $ kubectl create -f fluent-bit-04-role-binding-1.22.yaml
+   $ kubectl create -f fluent-bit-05-configmap.yaml
+   $ kubectl create -f fluent-bit-06-ds-minikube.yaml
+   ```
+6. Tunneling `my-nginx` service through minikube:
+   ```
+   $ minikube service my-nginx -n my-app
+   ```
+   example output:
+   ```
+   😿  service default/my-nginx has no node port
+   🏃  Starting tunnel for service my-nginx.
+   |-----------|----------|-------------|------------------------|
+   | NAMESPACE |   NAME   | TARGET PORT |          URL           |
+   |-----------|----------|-------------|------------------------|
+   | default   | my-nginx |             | http://127.0.0.1:62907 |
+   |-----------|----------|-------------|------------------------|
+   ```
+7. You could now curl the URL by
+   ```
+   curl http://127.0.0.1:62907
+   ```
+   and see response as follows:
+   ```
+   <!DOCTYPE html>
+   <html>
+   <head>
+   <title>Welcome to nginx!</title>
+   <style>
+   html { color-scheme: light dark; }
+   body { width: 35em; margin: 0 auto;
+   font-family: Tahoma, Verdana, Arial, sans-serif; }
+   </style>
+   </head>
+   <body>
+   <h1>Welcome to nginx!</h1>
+   <p>If you see this page, the nginx web server is successfully installed and
+   working. Further configuration is required.</p>
+   
+   <p>For online documentation and support please refer to
+   <a href="http://nginx.org/">nginx.org</a>.<br/>
+   Commercial support is available at
+   <a href="http://nginx.com/">nginx.com</a>.</p>
+   
+   <p><em>Thank you for using nginx.</em></p>
+   </body>
+   </html>
+   ```
+The deployed `nginx` pods will generate apache logs accordingly and Fluent Bit DaemonSet will collect, parse, filter and output logs to Data-Prepper backend. 
+8. By configuring `stdout` as sink, the Data-Prepper instance logs should now print out sample records as follows:
+```
+{"date":1.639425394678687E9,"log":"172.17.0.1 - - [13/Dec/2021:19:56:34 +0000] \"GET / HTTP/1.1\" 200 615 \"-\" \"curl/7.64.1\" \"-\"\n","stream":"stdout","time":"2021-12-13T19:56:34.6786871Z","kubernetes":{"pod_name":"my-nginx-5b56ccd65f-zfvtj","namespace_name":"my-app","pod_id":"2989f623-5a65-4caf-b33a-95154a04f53b","labels":{"pod-template-hash":"5b56ccd65f","run":"my-nginx"},"host":"minikube","container_name":"my-nginx","docker_id":"f827ae8f8ff66175da32776b3875fe10916378e89645288d415edfd22f060fdb","container_hash":"nginx@sha256:9522864dd661dcadfd9958f9e0de192a1fdda2c162a35668ab6ac42b465f0603","container_image":"nginx:latest"},"request":"/","auth":"-","ident":"-","response":"200","bytes":"615","clientip":"172.17.0.1","verb":"GET","httpversion":"1.1","timestamp":"13/Dec/2021:19:56:34 +0000"}
```

Comment:
I'm not seeing these when going through the instructions.

Reply:
Will doublecheck. I believe I have tried both data-prepper in ec2 and in docker

Reply:
OK. I ran this on my local Mac. I ran Data Prepper via Docker with the command above.

Reply:
Weird. Here is what I got:

```
docker run -p 2021:2021 -p 4900:4900 --name data-prepper -v ${PWD}/pipelines.yaml:/usr/share/data-prepper/pipelines.yaml opensearchproject/data-prepper:latest
WARNING: sun.reflect.Reflection.getCallerClass is not supported. This will impact performance.
2022-01-20T22:55:28,375 [main] INFO  com.amazon.dataprepper.pipeline.server.DataPrepperServer - Creating Data Prepper server with TLS
2022-01-20T22:55:28,690 [main] INFO        com.amazon.dataprepper.DataPrepper - Using /usr/share/data-prepper/pipelines.yaml configuration file
2022-01-20T22:55:28,710 [main] INFO  com.amazon.dataprepper.parser.PipelineParser - Building pipeline [grok-pipeline] from provided configuration
2022-01-20T22:55:28,711 [main] INFO  com.amazon.dataprepper.parser.PipelineParser - Building [http] as source component for the pipeline [grok-pipeline]
2022-01-20T22:55:28,718 [main] WARN  com.amazon.dataprepper.plugins.source.loghttp.HTTPSource - Creating http source without authentication. This is not secure.
2022-01-20T22:55:28,719 [main] WARN  com.amazon.dataprepper.plugins.source.loghttp.HTTPSource - In order to set up Http Basic authentication for the http source, go here: https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/http-source#authentication-configurations
2022-01-20T22:55:28,721 [main] INFO  com.amazon.dataprepper.parser.PipelineParser - Building buffer for the pipeline [grok-pipeline]
2022-01-20T22:55:28,726 [main] INFO  com.amazon.dataprepper.parser.PipelineParser - Building preppers for the pipeline [grok-pipeline]
2022-01-20T22:55:28,771 [main] INFO  com.amazon.dataprepper.parser.PipelineParser - Building sinks for the pipeline [grok-pipeline]
2022-01-20T22:55:28,772 [main] INFO  com.amazon.dataprepper.parser.PipelineParser - Building [stdout] as sink component
2022-01-20T22:55:28,775 [main] INFO  com.amazon.dataprepper.pipeline.Pipeline - Pipeline [grok-pipeline] - Initiating pipeline execution
2022-01-20T22:55:29,015 [main] WARN  com.amazon.dataprepper.plugins.source.loghttp.HTTPSource - Creating http source without SSL/TLS. This is not secure.
2022-01-20T22:55:29,016 [main] WARN  com.amazon.dataprepper.plugins.source.loghttp.HTTPSource - In order to set up TLS for the http source, go here: https://github.com/opensearch-project/data-prepper/tree/main/data-prepper-plugins/http-source#ssl
2022-01-20T22:55:29,356 [main] INFO  com.amazon.dataprepper.plugins.source.loghttp.HTTPSource - Started http source...
2022-01-20T22:55:29,356 [main] INFO  com.amazon.dataprepper.pipeline.Pipeline - Pipeline [grok-pipeline] - Submitting request to initiate the pipeline processing
2022-01-20T22:55:29,360 [main] INFO  com.amazon.dataprepper.pipeline.server.DataPrepperServer - Data Prepper server running at :4900
2022-01-20T22:55:32,364 [grok-pipeline-prepper-worker-1-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker -  grok-pipeline Worker: No records received from buffer
2022-01-20T22:55:42,194 [grok-pipeline-prepper-worker-1-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker -  grok-pipeline Worker: Processing 2 records from buffer
{"date":1.642719338200267E9,"log":"172.17.0.1 - - [20/Jan/2022:22:55:38 +0000] \"GET / HTTP/1.1\" 200 615 \"-\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.2 Safari/605.1.15\" \"-\"\n","stream":"stdout","time":"2022-01-20T22:55:38.2002665Z","kubernetes":{"pod_name":"my-nginx-5b56ccd65f-wsrnh","namespace_name":"my-app","pod_id":"8632d87f-0b45-47f9-9fc7-675cb48d3b17","labels":{"pod-template-hash":"5b56ccd65f","run":"my-nginx"},"host":"minikube","container_name":"my-nginx","docker_id":"4852d441b036964dbbc5c9e6d5cc0de21f825101a70165c549247596c23425ea","container_hash":"nginx@sha256:0d17b565c37bcbd895e9d92315a05c1c3c9a29f762b011a10c54a66cd53c9b31","container_image":"nginx:latest"},"request":"/","auth":"-","ident":"-","response":"200","bytes":"615","clientip":"172.17.0.1","verb":"GET","httpversion":"1.1","timestamp":"20/Jan/2022:22:55:38 +0000"}
{"date":1.642719341652019E9,"log":"172.17.0.1 - - [20/Jan/2022:22:55:41 +0000] \"GET / HTTP/1.1\" 200 615 \"-\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.2 Safari/605.1.15\" \"-\"\n","stream":"stdout","time":"2022-01-20T22:55:41.6520191Z","kubernetes":{"pod_name":"my-nginx-5b56ccd65f-wsrnh","namespace_name":"my-app","pod_id":"8632d87f-0b45-47f9-9fc7-675cb48d3b17","labels":{"pod-template-hash":"5b56ccd65f","run":"my-nginx"},"host":"minikube","container_name":"my-nginx","docker_id":"4852d441b036964dbbc5c9e6d5cc0de21f825101a70165c549247596c23425ea","container_hash":"nginx@sha256:0d17b565c37bcbd895e9d92315a05c1c3c9a29f762b011a10c54a66cd53c9b31","container_image":"nginx:latest"},"request":"/","auth":"-","ident":"-","response":"200","bytes":"615","clientip":"172.17.0.1","verb":"GET","httpversion":"1.1","timestamp":"20/Jan/2022:22:55:41 +0000"}
2022-01-20T22:55:51,164 [grok-pipeline-prepper-worker-1-thread-1] INFO  com.amazon.dataprepper.pipeline.ProcessWorker -  grok-pipeline Worker: Processing 1 records from buffer
{"date":1.642719347642312E9,"log":"172.17.0.1 - - [20/Jan/2022:22:55:47 +0000] \"GET / HTTP/1.1\" 200 615 \"-\" \"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.2 Safari/605.1.15\" \"-\"\n","stream":"stdout","time":"2022-01-20T22:55:47.6423123Z","kubernetes":{"pod_name":"my-nginx-5b56ccd65f-wsrnh","namespace_name":"my-app","pod_id":"8632d87f-0b45-47f9-9fc7-675cb48d3b17","labels":{"pod-template-hash":"5b56ccd65f","run":"my-nginx"},"host":"minikube","container_name":"my-nginx","docker_id":"4852d441b036964dbbc5c9e6d5cc0de21f825101a70165c549247596c23425ea","container_hash":"nginx@sha256:0d17b565c37bcbd895e9d92315a05c1c3c9a29f762b011a10c54a66cd53c9b31","container_image":"nginx:latest"},"request":"/","auth":"-","ident":"-","response":"200","bytes":"615","clientip":"172.17.0.1","verb":"GET","httpversion":"1.1","timestamp":"20/Jan/2022:22:55:47 +0000"}
```